firefox-ios/Client/Protocols/URLSession/URLSessionProtocol.swift

```diff
@@ -21,6 +23,17 @@ protocol URLSessionProtocol: Sendable {
     ) -> URLSessionDataTaskProtocol
 }
 
+/// Default implementation for bytes(for:) (for backward compatibility)
+/// Conforming types should implement their own bytes(for:) method.
+/// Attempting to use the default one here will yierld a runtime assertion failure.
+/// The throw is just to satisfy the async‑throws signature.
+extension URLSessionProtocol {
+    func bytes(for request: URLRequest) async throws -> (URLSession.AsyncBytes, URLResponse) {
+        assertionFailure("Fallback bytes(for:) called ! Conforming types their provide their own implementation")
```

Comment:
```suggestion
        assertionFailure("Fallback bytes(for:) called ! Conforming types provide their own implementation")
```

Reply:
Updated in https://github.com/mozilla-mobile/firefox-ios/pull/28284/commits/8e15ead46297af29d4e6b6b1cef5c8c5aab2f593